src/server/api/routers/club.ts

```diff
@@ -12,27 +15,46 @@ const byIdSchema = z.object({
 export const clubRouter = createTRPCRouter({
   byName: publicProcedure.input(byNameSchema).query(async ({ input, ctx }) => {
     const { name } = input;
-    try {
-      return await ctx.dbProvider.getClubsByName(name);
-    } catch (e) {
-      console.log(e);
-      return [];
-    }
+    const clubs = await ctx.db.query.club.findMany({
+      with: { contacts: true },
+    });
+
+    if (name === '') return clubs.map((c) => IClub.parse(c));
+
+    const fuse = new Fuse(clubs, { keys: ['name', 'description'] });
+    const results = fuse.search(name);
+    console.log({ results, name, clubs });
+    if (results.length === 0) return [];
+
+    const parsed = results.map((r) => IClub.parse(r.item));
+    if (parsed.length < 5) return parsed;
+    return parsed.slice(0, 5);
   }),
   byId: publicProcedure.input(byIdSchema).query(async ({ input, ctx }) => {
     const { id } = input;
     try {
-      return await ctx.dbProvider.getClubById(id);
+      const byId = await ctx.db.query.club.findFirst({
+        where: (club) => eq(club.id, id),
+        with: { contacts: true },
+      });
+
+      const first = IClub.parse(byId);
+      return first;
     } catch (e) {
       console.log(e);
```

Comment:
Minor but to keep consistent with line 57, it should be `console.error()` here as well.